mmv1/products/pubsub/Subscription.yaml

```
@@ -471,6 +471,7 @@ properties:
           A duration in seconds with up to nine fractional digits, terminated by 's'. Example: "3.5s".
         default_from_api: true
         diff_suppress_func: 'tpgresource.DurationDiffSuppress'
+    send_empty_value: true
```

I think you'll want allow_empty_object here as well, though it looks like the tests are passing, so maybe it's fine? Did you happen to try that out?

```suggestion
    send_empty_value: true
    allow_empty_object: true
```

---

Yeah the earlier BQ-related test failures seem unrelated and are passing now. Thanks for opening https://github.com/hashicorp/terraform-provider-google/issues/19683.

Those tests pass locally as well:
```
$ make testacc TEST=./google-beta/services/pubsub TESTARGS='-run=TestAccPubsubSubscription_pubsubSubscriptionPushBqExample'
==> Checking that code complies with gofmt requirements...
go vet
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta/services/pubsub -v -run=TestAccPubsubSubscription_pubsubSubscriptionPushBqExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccPubsubSubscription_pubsubSubscriptionPushBqExample
=== PAUSE TestAccPubsubSubscription_pubsubSubscriptionPushBqExample
=== CONT  TestAccPubsubSubscription_pubsubSubscriptionPushBqExample
--- PASS: TestAccPubsubSubscription_pubsubSubscriptionPushBqExample (30.70s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/pubsub 31.317s

$ make testacc TEST=./google-beta/services/pubsub TESTARGS='-run=TestAccPubsubSubscription_pubsubSubscriptionPushBqTableSchemaExample'
==> Checking that code complies with gofmt requirements...
go vet
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta/services/pubsub -v -run=TestAccPubsubSubscription_pubsubSubscriptionPushBqTableSchemaExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccPubsubSubscription_pubsubSubscriptionPushBqTableSchemaExample
=== PAUSE TestAccPubsubSubscription_pubsubSubscriptionPushBqTableSchemaExample
=== CONT  TestAccPubsubSubscription_pubsubSubscriptionPushBqTableSchemaExample
--- PASS: TestAccPubsubSubscription_pubsubSubscriptionPushBqTableSchemaExample (30.34s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/pubsub 30.958s
```

Similarly, the "VCR-test" was flaky as well but is passing now. However, the `terraform-google-conversion-build-and-unit-tests` failures show a permadiff with `"retryPolicy": nil,` ([example test case](https://github.com/GoogleCloudPlatform/terraform-google-conversion/blob/f8ac0b1b7915cd283393b1854d4d15d3f5d3876f/tfplan2cai/testdata/templates/example_pubsub_subscription.json#L30-L55) and [test code throwing the error](https://github.com/GoogleCloudPlatform/terraform-google-conversion/blob/7c35fd7478bb9702ba2c4e1ef325bc2f66488de8/tfplan2cai/test/read_test.go#L245)). I think that I'd need to submit a separate PR to fix those tests since they are in a different repo.

> I think you'll want allow_empty_object here as well, though it looks like the tests are passing, so maybe it's fine? Did you happen to try that out?

I didn't see that in the [field reference](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/field-reference.md) but I see it [here](https://github.com/GoogleCloudPlatform/magic-modules/blob/2d34af8bcf47b390ede98ace828c98e80cb3acd0/mmv1/api/type.rb#L93) and used elsewhere in `Subscription.yaml`. But yeah, the tests are passing with or without `allow_empty_object` and I'm not sure why, but it doesn't seem like adding `allow_empty_object` will hurt so I've added it.